examples/PyTorch/Inference/T5/main.py

```diff
@@ -0,0 +1,135 @@
+    for i in range(iters):
+        model(*tuple(inputs))
+    avg_time = (time.time() - tic) / iters
+    print("average time in {} iterations: {} seconds".format(iters, avg_time))
+
+    # profile start
+    cu_prof_start()
+    model(*tuple(inputs))
+    cu_prof_stop()
+    # profile end
+
+
+def disc_optimize(model, inputs, out_file: str):
+    torch_config = torch_blade.config.Config()
+    torch_config.enable_mlir_amp = False  # disable mix-precision
+    torch_config.enable_force_to_cuda = True
+
+    traced_model = torch.jit.trace(model.cuda(), inputs,
+                                   strict=False).cuda().eval()
+
+    torch._C._jit_pass_inline(traced_model._c.forward.graph)
+    torch._C._jit_pass_remove_dropout(traced_model._c)
+
+    with torch.no_grad(), torch_config:
+        # BladeDISC torch_blade optimize will return an optimized TorchScript
+        optimized_ts = torch_blade.optimize(traced_model,
+                                            allow_tracing=True,
+                                            model_inputs=tuple(inputs))
+    torch.jit.save(optimized_ts, out_file)
+
+
+def blade_trt_optimize(model, inputs, fp16: bool, out_file: str):
+    cfg = torch_blade.Config.get_current_context_or_new().clone()
+    cfg.optimization_pipeline = torch_blade.tensorrt.backend_name()
+    cfg.customize_onnx_opset_version = 12
+    cfg.enable_fp16 = fp16
+
+    traced_model = torch.jit.trace(model.cuda().eval(), inputs,
+                                   strict=False).cuda().eval()
+
+    with cfg, torch_blade.logging.logger_level_context('INFO'):
+        opt_model = torch_blade.optimize(traced_model,
+                                         False,
+                                         model_inputs=tuple(inputs))
+    torch.jit.save(opt_model, out_file)
+
+
+def run():
+    tokenizer = T5Tokenizer.from_pretrained("t5-base")
+    input_ids = tokenizer(
+        "Studies have been shown that owning a dog is good for you",
+        return_tensors="pt").input_ids  # Batch size 1
+    input_ids = input_ids.cuda()
+    mask = torch.ones_like(input_ids)
+    decoder_input_ids = tokenizer(
+        "Studies show that", return_tensors="pt").input_ids  # Batch size 1
+    decoder_input_ids = decoder_input_ids.cuda()
+    inputs = (input_ids, mask, decoder_input_ids)
+
+    model = T5Model.from_pretrained("t5-base", torchscript=True).eval().cuda()
+    traced_model_amp = trace_model(model, inputs, True).eval().cuda()
+
+    # Run naive torch.
+    print("Naive PyTorch.")
+    model = traced_model_amp
+    evaluate_torch(model, inputs)
+
+    # Run BladeDISC optimization.
+    print("BladeDISC Optimization.")
+    disc_optimize(traced_model_amp, inputs, 't5-base_amp.disc.pt')
+    model = torch.jit.load('t5-base_amp.disc.pt').cuda().eval()
+    evaluate_torch(model, inputs)
+
+    # Run TorchBlade-TensorRT optimization.
+    print("TorchBlade-TensorRT Optimization.")
+    blade_trt_optimize(model, inputs, True, 't5-base_amp.trt.pt')
```

Comment:
Seems like it's running blade_trt optimization on a disc-optimized model, which was loaded at L124.

Reply:
Thanks for your comment. I have removed the TRT part because I found TRT optimization generates wrong results.